nexus/reconfigurator/planning/src/blueprint_editor/sled_editor.rs

```diff
@@ -745,6 +791,189 @@ impl ActiveSledEditor {
         Ok(())
     }
 
+    /// Update a sled's mupdate override field based on the mupdate override
+    /// provided by inventory.
+    pub fn ensure_mupdate_override(
+        &mut self,
+        inv_mupdate_override_id: Option<MupdateOverrideUuid>,
+        pending_mgs_update: Entry<'_, PendingMgsUpdate>,
+        noop_sled_info: NoopConvertSledInfoMut<'_>,
+    ) -> Result<EnsureMupdateOverrideAction, SledEditError> {
+        match (inv_mupdate_override_id, *self.remove_mupdate_override.value()) {
+            (Some(inv_override), Some(bp_override))
+                if inv_override == bp_override =>
+            {
+                // If the inventory and blueprint overrides are the same, the
+                // sled agent hasn't yet removed the override. Nothing to do at
+                // the moment.
+                Ok(EnsureMupdateOverrideAction::NoAction {
+                    mupdate_override: Some(inv_override),
+                })
+            }
+            (Some(inv_override), bp_override) => {
+                // Inventory says there's an override in place, but the
+                // blueprint doesn't (or has a different override in place).
+                // This means that a MUPdate happened since we last did
+                // blueprint planning.
+                //
+                // Set the blueprint's remove_mupdate_override.
+                self.set_remove_mupdate_override(Some(inv_override));
+                // Also update the cached value inside `noop_sled_info`.
+                if let NoopConvertSledInfoMut::Ok(mut info) = noop_sled_info {
+                    use NoopConvertSledIneligibleReason::*;
+
+                    match &mut info.status {
+                        NoopConvertSledStatus::Ineligible(
+                            MupdateOverride { mupdate_override_id, .. },
+                        ) => {
+                            *mupdate_override_id = inv_override;
+                        }
+                        NoopConvertSledStatus::Ineligible(_) => {
+                            // Some other reason -- sled remains ineligible.
+                        }
+                        NoopConvertSledStatus::Eligible(eligible) => {
+                            // Transition to Eligible with the new override.
+                            let zones = mem::replace(
```

Comment:
Thoughts on `mem::take()` instead of `mem::replace()`? I can never decide whether it feels too magical.

Reply:
In this case, yeah, `mem::take` makes sense -- we're replacing it immediately anyway.